src/operator/nn/pooling-inl.h

```diff
@@ -132,19 +132,21 @@ class PoolingOp {
     using namespace mshadow;
     Stream<xpu> *s = ctx.get_stream<xpu>();
     const TShape& ishape = in_data.shape_;
+    TShape kernel = param_.kernel;
     TShape padding = param_.pad;
+    TShape stride = param_.stride;
     if (param_.global_pool) {
-      for (index_t i = 0; i < padding.ndim(); i++) {
+      kernel = TShape(ishape.data()+ishape.ndim()-param_.kernel.ndim(), ishape.data()+ishape.ndim());
```

Comment:
@sxjscience 
It still does not work even though kernel is set accordingly. By the way, I do not think the original implementation does wrong here, as it also considers the global_pool and resets the kernel.  
Also I am not familiar with the source codes and the codes are in such a mass that I failed to find the definition of ```TShape```. Would you please show me where ```TShape``` is defined?